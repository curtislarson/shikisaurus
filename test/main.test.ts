import { assert, assertSnapshot } from "./test-deps.ts";

Deno.test("main", { sanitizeExit: false, sanitizeOps: false, sanitizeResources: false }, async (t) => {
  await t.step("run and query urls", async (t) => {
    const res = new Deno.Command(new URL("../server.ts", import.meta.url).pathname, {
      stdout: "piped",
      stderr: "inherit",
    }).spawn();

    const chunk = new TextDecoder().decode((await res.stdout.getReader().read()).value);

    assert(chunk.startsWith("Now serving"));

    const val = await fetch("http://localhost:8000/shikisaurus", {
      method: "POST",
      body: JSON.stringify({ code: "console.log(\"hello world\");" }),
    }).then((r) => r.text());

    await assertSnapshot(t, val);

    const githubVal = await fetch(
      "http://localhost:8000/shikisaurus/github?url=https%3A%2F%2Fgithub.com%2Fquackware%2Fquackware%2Fblob%2Fmaster%2F.editorconfig%23L4-L5",
      {
        method: "GET",
      },
    ).then((r) => r.text());

    await assertSnapshot(t, githubVal);
  });
});
