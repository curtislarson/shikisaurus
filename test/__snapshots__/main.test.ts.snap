export const snapshot = {};

snapshot[`main > run and query urls 1`] = `'<pre class="shiki" style="background-color: #282A36"><code><span class="line"><span style="color: #F8F8F2">console.</span><span style="color: #50FA7B">log</span><span style="color: #F8F8F2">(</span><span style="color: #E9F284">&quot;</span><span style="color: #F1FA8C">hello world</span><span style="color: #E9F284">&quot;</span><span style="color: #F8F8F2">);</span></span></code></pre>'`;

snapshot[`main > run and query urls 2`] = `
'<pre class="shiki" style="background-color: #282A36"><code><span class="line"><span style="color: #F8F8F2">end_of_line </span><span style="color: #FF79C6">=</span><span style="color: #F8F8F2"> lf</span></span>
<span class="line"><span style="color: #F8F8F2">insert_final_newline </span><span style="color: #FF79C6">=</span><span style="color: #F8F8F2"> </span><span style="color: #BD93F9">true</span></span></code></pre>'
`;
